### PR TITLE
pull1

### PR DESCRIPTION
**Pull request by georgef21**

### CODE REVIEW

**Comment by antweiss** [comment]
@YonatanItzhak please review

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  #config.vm.synced_folder ".", "/var/webapp"
+
+
+# provision settings
+
+ config.vm.provision "shell", inline: <<-SHELL
+  sudo apt-get update
+  sudo apt install python3.8-venv -y
+  sudo apt install python3-pip -y
```
Comment (YonatanItzhak) [inline Vagrantfile]: No need to install python here because python is also installed in the deployment bash script

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  sudo apt install curl
+  curl -LJO https://raw.githubusercontent.com/georgef21/Flask-vagrant-project/main/webapp.py
+  curl -LJO https://raw.githubusercontent.com/georgef21/Flask-vagrant-project/main/deployment
```
Comment (YonatanItzhak) [inline Vagrantfile]: This does not install the resources there is no repo named 'Flask-vagrant-project', you need to change the URL here and for webapp.py
```suggestion
  curl -LJO https://raw.githubusercontent.com/georgef21/vagrant-flask-exercise/main/deployment
```

File: deployment [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+source venv/bin/activate
+
+pip install Flask
+
+python -m flask --version
+
+export FLASK_APP=webapp.py
+
+flask run
```
Comment (YonatanItzhak) [inline deployment]: I see in the output     `default:  * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)`
But in fact when trying to access this port I get this error:
ERR_CONNECTION_RESET 
are you able to see the 'Hello world!' web page?

File: README1.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+in order to run the program on linux, instalation of vagrant and virtualbox is requierd installing vagrant:
+
+sudo apt update sudo apt install virtualbox
```
Comment (YonatanItzhak) [inline README1.md]: Please make sure that commands in README are clear, looks like you forgot to go down a line here, and in line 5 and 9

File: Vagrantfile (hunk 2 above)
Comment (antweiss) [reply Vagrantfile]: @georgef21 this fails for me too. Please fix

File: deployment (hunk 3 above)
Comment (antweiss) [reply deployment]: @georgef21 - Yonatan has a point here. Please research.

File: Vagrantfile (hunk 2 above)
Comment (antweiss) [reply Vagrantfile]: Actually you don't even need to download anything. The application files are right here in the repo next to Vagrantfile.

Comment (antweiss) [reply Vagrantfile]: just use shell provisioner with `path` and not `inline`

File: README1.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+curl -O https://releases.hashicorp.com/vagrant/2.2.9/vagrant_2.2.9_x86_64.deb sudo apt install ./vagrant_2.2.9_x86_64.deb
+
+after the instalation a directory must be created and the vagrant file provided in github must be copied there using the terminal
+
+mkdir vm sudo mv Vagrantfile vm
```
Comment (antweiss) [inline README1.md]: Why not just run vagrant from the directory where Vagrantfile already is? There's no need to move it to another directory

File: Vagrantfile [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+ 
+ 
+ #network settings
+ 
+  config.vm.network "forwarded_port", guest: 80, host: 8080
+
+  config.vm.network "forwarded_port", guest: 5000, host: 5000
+
+
+  # config.vm.network "private_network", ip: :127.0.0.1:5000"
```
Comment (YonatanItzhak) [inline Vagrantfile:45]: I think you can get rid of the commented lines

File: automate [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (YonatanItzhak) [inline automate:1]: This is nice but it only supported only on linux machines right? not macOS or windows.. should be mentioned in my opinion

File: webapp.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+from flask import Flask
+app = Flask(__name__)
+
+@app.route("/hello")
```
Comment (YonatanItzhak) [inline webapp.py:4]: If I go to localhost:5000 I get 404 not found, only localhost:5000/hello works
this is ok, but after the flask app is served, the stdout says: "Running on http://10.0.2.15:5000/"
So I'd expect this address to work while it does not actually exist, so maybe remove the "/hello" and make it only "/"
Or have both :)

**Comment by YonatanItzhak** [comment]
Looks better now, the app works for me

**Pull request closed without merging**